tests/DCPS/GroupPresentation/SubscriberListener.h

```diff
@@ -67,6 +67,7 @@ class SubscriberListenerImpl
 
   DDS::Subscriber_var subscriber_;
   bool  verify_result_;
+  ACE_Recursive_Thread_Mutex listener_lock_;
```

Comment:
Does this really need to be recursive? I only see it being locked in on_data_on_readers(), which I wouldn't think calls back to itself?

Reply:
Probably not, but I was following a pattern, since pretty much all the other locks are recursive, I just went with it.